src/storage/src/hummock/compactor/context.rs

```diff
@@ -102,7 +102,7 @@ impl CompactorContext {
         }
     }
 
-    pub fn acquire_task_quota(&self, parallelism: u32) -> bool {
+    pub fn acquire_task_quota(&self, parallelism: u32) -> Option<ReleaseGuard> {
```

Comment:
1. We don't need to store `max_task_parallelism` in compactor conext becuase it will only be used in the compactor event loop.
2. `max_task_parallelism` doesn't need to be a atomic because it will only be accessed in a single thread. A local var in the event loop is sufficient.
3. due to 1 and 2, `acquire_task_quota`, `release_task_quota` and `get_free_quota` can be removed.